model/common/src/icon4py/model/common/model_options.py

```diff
@@ -5,12 +5,60 @@
 #
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
+import functools
+import typing
 
-from gt4py.eve.utils import FrozenNamespace
+import gt4py.next as gtx
+from gt4py._core.definitions import (
+    is_scalar_type,  # TODO(havogt): Should this function be public API?
+)
+from gt4py.next import backend
 
 
-class RayleighType(FrozenNamespace[int]):
-    #: classical Rayleigh damping, which makes use of a reference state.
-    CLASSIC = 1
-    #: Klemp (2008) type Rayleigh damping
-    KLEMP = 2
+def dict_values_to_list(d: dict[str, typing.Any]) -> dict[str, list]:
+    return {k: [v] for k, v in d.items()}
+
+
+def setup_program(
+    backend: backend.Backend,
+    program: gtx.program,
+    constant_args: dict | None = None,
+    variants: dict | None = None,
+    horizontal_sizes: dict | None = None,
+    vertical_sizes: dict | None = None,
+    offset_provider: dict | None = None,
+) -> typing.Callable[..., None]:
+    """
+    This function processes pre-compiled args and feeds some to the gt4py `compile` function.
+    Args:
+        - backend: pre-set backend at run time,
```

Comment:
I don't understand what you want to express here, maybe just
```suggestion
        - backend: GT4Py backend,
```